### ISSUE TITLE
`while something` breaks type mismatch detection for `something`

### ISSUE THREAD

**Issue by Molkree** [microsoft/pylance-release#2105]
## Environment data

-   Language Server version: v2021.11.2
-   OS and version: Win10
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.10
## Expected behaviour

Errors are being reported inside the `while` loop and after the loop.

## Actual behaviour

Type mismatch errors for `something` object are "suppressed" in the loop AND after the loop. Maybe other errors too? `something` can be a container (with which I have discovered it) but even a simple `str` will (not) work.
## Logs

This is when uncommenting the `string` example (see below) in an otherwise fully commented out file. Don't think the logs are very helpful here.
```
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
[FG] parsing: c:\Users\mrmol\Downloads\experiments.py (0ms)
[FG] binding: c:\Users\mrmol\Downloads\experiments.py (0ms)
Background analysis message: getSemanticTokens delta
[BG(1)] getSemanticTokens delta previousResultId:1637702644443 at c:\Users\mrmol\Downloads\experiments.py ...
[BG(1)]   parsing: c:\Users\mrmol\Downloads\experiments.py (1ms)
[BG(1)]   binding: c:\Users\mrmol\Downloads\experiments.py (0ms)
[BG(1)] getSemanticTokens delta previousResultId:1637702644443 at c:\Users\mrmol\Downloads\experiments.py (2ms)
Background analysis message: analyze
[BG(1)] analyzing: c:\Users\mrmol\Downloads\experiments.py ...
[BG(1)]   checking: c:\Users\mrmol\Downloads\experiments.py (10ms)
[BG(1)] analyzing: c:\Users\mrmol\Downloads\experiments.py (10ms)
Background analysis message: resumeAnalysis
```

## Code Snippet / Additional information

This is with strict type checking.

```python
my_set = set([1, 3, 2])
my_list = list(my_set)

# if you uncomment the line below you will see `Operator "-=" not supported for types "set[int]"
# and "list[int]"` and `Type of "my_set" is unknown` as a consequence
# This is our desired error warning
# my_set -= my_list
while my_set:
    my_set -= my_list  # no warning
    my_set -= "string"  # no warning

my_set -= my_list  # no warning even after the loop
new_set = set([1]) - [1]  # warnings work for other objects

# example above also works if you swap `list` and `set`

# and it also works with strings
string = "5"
while string:
    string -= 5  # no warning
string -= 5  # no warning
```

**Comment by erictraut** [comment]
Thanks for the bug report.

Good news: this problem has already been fixed. The next version of pylance will include the fix.

![Screen Shot 2021-11-23 at 1 41 19 PM](https://user-images.githubusercontent.com/7040122/143133579-041393da-8f7e-42b9-8d98-c7a72cea1d73.png)

**Comment by bschnurr** [comment]
This issue has been fixed in version 2021.12.0, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2021120-2-december-2021)
